src/App.js

```diff
@@ -2,27 +2,31 @@ import React, { useEffect, useState } from "react";
 import AddTodoForm from "./AddTodoForm";
 import TodoList from "./TodoList";
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem("SavedTodoList")) ?? []
-  );
-
-  useEffect(() => {
-    localStorage.setItem("SavedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
-  return [todoList, setTodoList];
-}
-
 function App() {
   const [todoList, setTodoList] = useSemiPersistentState();
   const addTodo = (newTodo) => {
     setTodoList([...todoList, newTodo]);
   };
+
+  function useSemiPersistentState() {
+    const [todoList, setTodoList] = useState(
+      JSON.parse(localStorage.getItem("SavedTodoList")) ?? []
+    );
+
+    useEffect(() => {
+      localStorage.setItem("SavedTodoList", JSON.stringify(todoList));
+    }, [todoList]);
+    return [todoList, setTodoList];
+  }
```

Comment:
Nitpick:
I'd recommend keeping this `useSemiPersistentState` hook outside of the `App` since it's a custom hook that isn't concerned with the `App` functionality.